packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/navigator-button.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * WordPress dependencies
+ */
+
+import { __experimentalNavigatorButton as NavigatorButton } from '@wordpress/components';
+import { useEntityRecords } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { PRELOADED_NAVIGATION_MENUS_QUERY } from './constants';
+
+export default function SidebarNavigationScreenNavigationMenuButton( {
+	children,
+	...props
+} ) {
+	const { records: navigationMenus, isResolving: isLoading } =
+		useEntityRecords(
+			'postType',
+			`wp_navigation`,
+			PRELOADED_NAVIGATION_MENUS_QUERY
+		);
+
+	const hasNavigationMenus = !! navigationMenus?.length;
+	const hasSingleNavigationMenu = navigationMenus?.length === 1;
+	const firstNavigationMenu = navigationMenus?.[ 0 ];
+
+	const showNavigationScreen = process.env.IS_GUTENBERG_PLUGIN
+		? hasNavigationMenus
+		: false;
+
+	// If there is a single menu then we can go directly to that menu.
+	// Otherwise we go to the navigation listing screen.
+	const path = hasSingleNavigationMenu
```

Comment:
I'm not sure it's wise to do this (change the path depending on the number of menus). This creates an issue because now the "parent" of the navigation is "hidden". so for instance if you go to "navigation" and you click "back" you don't end up in the root place.

I wonder if the best solution is to instead of doing this, change the behavior within the `/navigation` screen to actually render the menu directly when there's one.

Reply:
That's a good point. I don't want to expose the need to duplicate that logic. Thank you for spotting it.

I think I had that functionality initially but the issue was that I needed a dedicated route for a single Navigation. That is a requirement for Nav Focus mode.

That said, if we accept the above, then we could _also_ make the navigation listing route conditionally display 

- the list of menus
- a single menu if there's only one

As long as we still retain the single route that would be good. 

I'll raise an Issue

Reply:
https://github.com/WordPress/gutenberg/issues/51561